lib/oplog/oplogEntry.go

```diff
@@ -125,21 +99,29 @@ func (op *oplogEntry) ChangedFields() []string {
 		// { $v: 1, $set: { "a": 10, "b": 20 }, $unset: { "c": true } }
 
 		fields := []string{}
-		for operationKey, operation := range op.Data {
+		elements, err := op.Data.Elements()
+		if err != nil {
+			metricUnprocessableChangedFields.Inc()
+			log.Log.Errorw("Oplog data for non-replacement v1 update failed to unmarshal",
+				"op", op, "error", err)
+			return []string{}
```

Comment:
Should bubble up an error rather than returning a zero value?

Reply:
The existing function this is in doesn't currently return an error, so I'm trying to follow the existing behavior here as much as possible, minimizing how much is changed in this PR.  Making this return an error will be a bigger refactor, which certainly may be a good idea, but my thought was that it might be beyond the scope of this PR.  Let me know if you think otherwise.

Reply:
How did the old version behave when it hits this case? Does the whole unmarshall operation fail, or does it return the elements that it was able to process successfully? Want to make sure there's not a subtle difference of behavior here since this [operation](https://pkg.go.dev/go.mongodb.org/mongo-driver/bson#Raw.Elements) attempts to return a partial result (that's being ignored here)

Reply:
Yeah, that's a good point and it's not documented how the bson.Unmarshal handles partially corrupted data, but it seems it would be safe to assume that corrupt data would cause an error, so in this case maybe we do want to handle that, though it will take some refactoring.

Reply:
Looking at it again, returning an empty set on error doesn't make much difference in the end, but in this case refactoring to include an error return wouldn't be too hard since the parent of the caller returns an error.

Reply:
I did add an error that will bubble up to the level where the original unmarshal was